### PR TITLE
Miscellaneous updates

### PR DESCRIPTION
**Pull request by darconeous**
This pull requests contains miscellaneous minor fixes and improvements:

* any-to: Allow any_to_string to always work.
* socket-utils: Allow the explicit specification of a TCP socket.
* socket-utils: Use constant macros for setting baud rate.
* Fix various warnings and ignored files.
* connman-plugin: Updates to get the plugin compiling on Android
* connman-plugin: Don't include mcheck.h

### CODE REVIEW

File: src/util/any-to.cpp [hunk 1]
```diff
@@ -37,18 +37,36 @@ using namespace nl;
...
+	nl::Data ret;
 
 	if (value.type() == typeid(std::string)) {
 		std::string key_string = boost::any_cast<std::string>(value);
-		net_key = nl::Data(
-		    (const uint8_t*)key_string.c_str(), key_string.length());
+		uint8_t data[key_string.size()/2];
+
+		int length = parse_string_into_data(data,
+											key_string.size()/2,
```
Comment (abtink) [inline src/util/any-to.cpp]: nit: the spacing seems weird here (prob spaces vs tabs?)

File: src/util/any-to.cpp [hunk 2]
```diff
@@ -223,9 +243,33 @@ std::string any_to_string(const boost::any& value)
...
 	} else if (value.type() == typeid(nl::Data)) {
-		ret = "<nl::Data>";
+		nl::Data data = boost::any_cast<nl::Data>(value);
+		ret = std::string(data.size()*2,0);
+
+		// Reserve the zero termination
+		ret.reserve(data.size()*2+1);
+
+		encode_data_into_string(data.data(),
+								data.size(),
```
Comment (abtink) [inline src/util/any-to.cpp]: nit: spacing weirdness...

File: src/util/any-to.cpp (hunk 1 above)
Comment (darconeous) [inline src/util/any-to.cpp]: Will have a look.

**Comment by darconeous** [comment]
I'm going to clean those up in a later commit since I want to get these changes in ASAP.

**Pull request merged**